File: go/raindrops/raindrops.go
package raindrops

import "strconv"

// Convert Number to Pling Plang Plong or Number
func Convert(i int) string {
	var res string

	if i%3 == 0 {
		res += "Pling"
	}
	if i%5 == 0 {
		res += "Plang"
	}
	if i%7 == 0 {
		res += "Plong"
	}
	if res == "" {
		res += strconv.Itoa(i)
	}

	return res
}

File: elixir/raindrops/raindrops.exs
defmodule Raindrops do
  @doc """
  Returns a string based on raindrop factors.

  - If the number contains 3 as a prime factor, output 'Pling'.
  - If the number contains 5 as a prime factor, output 'Plang'.
  - If the number contains 7 as a prime factor, output 'Plong'.
  - If the number does not contain 3, 5, or 7 as a prime factor,
    just pass the number's digits straight through.
  """
  @spec convert(pos_integer) :: String.t
  def convert(number) do
    [
      raindrop(number, 3, "Pling"),
      raindrop(number, 5, "Plang"),
      raindrop(number, 7, "Plong"),
    ]
    |> Enum.join
    |> output(to_string(number))
  end

  defp raindrop(number, factor, word) do
    case rem(number, factor) do
      0 -> word
      _ -> ""
    end
  end

  defp output("", through), do: through
  defp output(raindrops, _), do: raindrops
end
